src/pybind/mgr/dashboard/frontend/src/app/shared/directives/auth-storage.directive.ts

```diff
@@ -0,0 +1,93 @@
+import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
+
+import _ from 'lodash';
+
+import { Permission, Permissions } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+
+@Directive({
+  selector: '[cdAuthStorage]'
```

Comment:
Sorry if I already asked you to change it to this name, but I was wondering if `cdAuthStorage` is really meaningful.

What about `cdScope`? That has the cd namespace prefix (of Ceph-Dashboard), and with Scope to highlight the fact that the condition to check is about authorization scopes.

Reply:
Hello @epuertat, cdScope sounds good but I'm wondering if it deals with all permission scopes or just storage permission scopes

Reply:
Also, Should we rename the directive to scope directive?

Reply:
@nSedrickm The 'storage' part in the original service is totally misleading. It simply refers to the fact that the permissions were 'stored' in the browser LocalStorage, and hence the 'storage' part, but it has nothing to do with Ceph storage.

So anything like: `cd` (from Ceph Dashboard, as Angular recommends prefixing directives with namespaces in order to avoid collisions... `ng` for Angular, etc.) + `Scope` or `Feature` of `Component` or whatever.

The whole idea of this is that a developer that adds a new piece of code just has to 'tag' that code with the component/scope/feature (whichever you prefer) it belongs to, and the visibility of that code is automatically managed by this directive + auth storage service (which we should rename in the future too ;-) )

Reply:
Thanks